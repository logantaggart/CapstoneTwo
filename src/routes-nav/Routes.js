import React from 'react'
import { Route, Switch } from 'react-router-dom'
import PrivateRoute from './PrivateRoute'

import Homepage from '../homepage/Homepage'

import LoginForm from '../auth/LoginForm'
import SignupForm from '../auth/SignupForm'

import Currency from '../currency/Currency'

import Watchlist from '../watchlist/Watchlist'
import Profile from '../profile/Profile'

function Routes({ login, signup, retrieveWatchlist }) {
    return (
        <div className="pt-5">
            <Switch>

                <Route exact path="/">
                    <Homepage retrieveWatchlist={retrieveWatchlist} />
                </Route>

                <Route exact path="/login">
                    <LoginForm login={login} />
                </Route>

                <Route exact path="/signup">
                    <SignupForm signup={signup} />
                </Route>

                <PrivateRoute exact path="/currency">
                    <Currency />
                </PrivateRoute>

                <PrivateRoute exact path='/watchlist'>
                    <Watchlist retrieveWatchlist={retrieveWatchlist} />
                </PrivateRoute>

                <PrivateRoute exact path="/profile">
                    <Profile />
                </PrivateRoute>

            </Switch>
        </div >
    )
}

export default Routes