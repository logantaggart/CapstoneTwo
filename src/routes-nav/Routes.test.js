import React from 'react'
import { MemoryRouter } from 'react-router'

import { render } from '@testing-library/react'

import UserProvider from '../testUtils'

import Routes from './Routes'

it('renders without crashing', () => {
    render(
        <MemoryRouter>
            <UserProvider>
                <Routes />
            </UserProvider>
        </MemoryRouter>
    )
})

it('matches snapshot', () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <Routes />
            </UserProvider>
        </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot()
})