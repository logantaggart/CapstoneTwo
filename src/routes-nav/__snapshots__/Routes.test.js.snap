// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="pt-5"
  >
    <div
      class="homepage"
    >
      <div
        class="container text-center"
      >
        <h1
          class="mb-3 font-weight-bold logo"
        >
          Cryptic
        </h1>
        <div>
          <form>
            <input
              class="form-control form-control-lg flex-grow-1"
              name="cryptoname"
              placeholder="Please Enter A Cryptocurrency Name (Ex: Ethereum)"
              required=""
              value=""
            />
            <br />
            <select
              class="form-control form-control-sm flex-grow-1"
              name="convertname"
            >
              <option
                value="USD"
              >
                USD ($)
              </option>
              <option
                value="EUR"
              >
                EUR (€)
              </option>
              <option
                value="JPY"
              >
                JPY (¥)
              </option>
              <option
                value="GPY"
              >
                GBP (£)
              </option>
              <option
                value="AUD"
              >
                AUD (A$)
              </option>
              <option
                value="CAD"
              >
                CAD (C$)
              </option>
              <option
                value="CHF"
              >
                CHF (Fr)
              </option>
              <option
                value="CNY"
              >
                CNY (¥)
              </option>
              <option
                value="HKD"
              >
                HKD (HK$)
              </option>
              <option
                value="NZD"
              >
                NZD (NZ$)
              </option>
              <option
                value="SEK"
              >
                SEK (Kr)
              </option>
              <option
                value="KRW"
              >
                KRW (₩)
              </option>
              <option
                value="SGD"
              >
                SGD (S$)
              </option>
              <option
                value="NOK"
              >
                NOK (Kr)
              </option>
              <option
                value="MXN"
              >
                MXN ($)
              </option>
              <option
                value="INR"
              >
                INR (₹)
              </option>
              <option
                value="RUB"
              >
                RUB (₽)
              </option>
              <option
                value="ZAR"
              >
                ZAR (R)
              </option>
              <option
                value="TRY"
              >
                TRY (₺)
              </option>
              <option
                value="BRL"
              >
                BRL (R$)
              </option>
            </select>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
